### PR TITLE
server: raise error when bootstrap with a zero store-id

### LINKED ISSUE #13011: server: raise error when bootstrap with a zero store-id
## Development Task

We found that [pd can return 0 when alloc id](https://github.com/tikv/pd/blob/813c88199d8eb531f9ae303ced095ff12e793337/server/id/id.go#L65), usually because of a failure to update base.

A zero store-id is weird, I think we should at least return a error rather than debug here.

### PR DESCRIPTION
**Pull request by CalvinNeo**
Signed-off-by: CalvinNeo <calvinneo1995@gmail.com>

### What is changed and how it works?

Issue Number: Close #13011

What's Changed:

We found that a zero-store id can be generated when alloc id.

A zero store-id is weird, I think we should at least return a error rather than debug here. 

```commit-message
raise error when bootstrap with a zero store-id
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by BusyJay** [comment]
I don't get it. If error is returned, it should check the error and return RPC error right?

**Comment by CalvinNeo** [comment]
> I don't get it. If error is returned, it should check the error and return RPC error right?

If pd meets a error in `rebaseLocked`, it will return a err. However, it caller `Alloc` returns `id = 0` instead of forward this err as an RPC error. I don't know why pd does so, however, I think we should print out at lease a error in TiKV side.

**Comment by BusyJay** [comment]
> If pd meets an error in `rebaseLocked`, it will return a err. However, its caller `Alloc` returns `id = 0` instead of forward this err as an RPC error. I don't know why pd did so, however, I think we should print out at least a error on TiKV side when such thing happen.

I think it deserves an issue in PD repository. It seems a bug. /cc @nolouch

File: src/server/node.rs [hunk 1]
```diff
@@ -167,7 +167,11 @@ where
         let mut store_id = self.check_store(&engines)?;
         if store_id == INVALID_ID {
             store_id = self.alloc_id()?;
-            debug!("alloc store id"; "store_id" => store_id);
+            if store_id == 0 {
```
Comment (BusyJay) [inline src/server/node.rs:170]: It should return error instead. `store_id` should not be 0.

Comment (CalvinNeo) [reply src/server/node.rs:170]: addressed

**Comment by BusyJay** [comment]
https://github.com/tikv/tikv/blob/master/components/raftstore-v2/src/bootstrap.rs#L75 also needs to be checked. How about changing the implementation of `alloc_id` and check if the returned value is zero?

**Review by BusyJay — APPROVED** [review]

**Comment by CalvinNeo** [comment]
> Where is the PD issue?

After dig a little deeper, I found may be the zero store id is not caused by pd's returning value.

**Review by breezewish — APPROVED** [review]

**Pull request merged**